src/EssentialSigner/index.ts

```diff
@@ -1,30 +1,45 @@
 import {
-  BlockTag,
   Provider,
   TransactionRequest,
   TransactionResponse,
 } from '@ethersproject/abstract-provider';
+import { ExternallyOwnedAccount } from '@ethersproject/abstract-signer';
 import { Deferrable, defineReadOnly } from '@ethersproject/properties';
-import { logger, Signer } from 'ethers';
-import { Bytes, Logger, resolveProperties } from 'ethers/lib/utils';
+import { BigNumberish, logger, Signer, Wallet } from 'ethers';
+import { Bytes, Logger } from 'ethers/lib/utils';
 
-import { EIP712Domain, EIP712StructField } from './messageSigner';
+import EssentialForwarder from '../abi/EssentialForwarder.json';
+import {
+  EIP712Domain,
+  EIP712StructField,
+  signMetaTxRequest,
+} from './messageSigner';
+
+export interface EssentialOverrides {
+  authorizer: string;
+  nftContract: string;
+  nftChainId: BigNumberish;
+  nftTokenId: BigNumberish;
+}
 
-export class EssentialSigner extends Signer {
+export class EssentialSigner extends Signer implements ExternallyOwnedAccount {
   readonly address: string;
+  readonly privateKey: string;
   readonly provider: Provider;
   readonly relayerUri: string;
 
   constructor(
     address: string,
-    provider: Provider,
+    provider?: Provider,
+    wallet?: Wallet,
     // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
     relayerUri: string = process.env.RELAYER_URI!,
   ) {
-    logger.checkNew(new.target, EssentialSigner);
+    // logger.checkNew(new.target, EssentialSigner);
     super();
     defineReadOnly(this, 'address', address);
-    defineReadOnly(this, 'provider', provider);
+    provider && defineReadOnly(this, 'provider', provider);
+    wallet && defineReadOnly(this, 'privateKey', wallet.privateKey);
```

Comment:
our solution needs to handle both a Web3Provider as well as provate key based signing. when we use our burner wallet approach we use the private key string to sign, but we still want to support manual signing.

I get a little lost between providers / signers / wallets here, and would like to simplify this bit.